source/Firmata/UwpFirmata.h

```diff
@@ -374,128 +374,24 @@ public ref class UwpFirmata sealed
 
-    ///<summary>
-    ///When used with std::bind, this allows the Firmata library to invoke the function in the standard way (non-member type) while we redirect it to an object reference.
-    ///</summary>
-    static inline
-    void
-    sysexInvoke(
-        UwpFirmata ^caller_,
-        uint8_t command_,
-        uint8_t argc_,
-        uint8_t *argv_
-    )
-    {
-        DataWriter ^writer = ref new DataWriter();
-        uint8_t i, len;
-
-        //Firmata does not handle capability responses in the typical way (separating bytes), so a special case is needed 
-        if( command_ == static_cast<uint8_t>( SysexCommand::CAPABILITY_RESPONSE ) )
-        {
-            for( i = 0; i < argc_; ++i )
-            {
-                writer->WriteByte( argv_[i] );
-            }
-            caller_->PinCapabilityResponseReceived( caller_, ref new SysexCallbackEventArgs( command_, writer->DetachBuffer() ) );
-            return;
-        }
-
-        /*
-         * data will be replied as 2 7-bit bytes for every actual byte. So we're going to reuse
-         *  the same memory space, since we can combine the two bytes back together.
-         */
-
-        //should never happen, but we'll correct for it just in case
-        if( argc_ % 2 == 1 ) --argc_;
-
-        //reassemble all the bytes (which were split into two seven-bit bytes) back into one byte each
-        for( i = 0, len = 0; i < argc_; i += 2, ++len )
-        {
-            argv_[len] = argv_[i] | ( argv_[i + 1] << 7 );
-        }
-        argv_[len] = 0;
-
-        if( command_ == static_cast<uint8_t>( SysexCommand::I2C_REPLY ) )
-        {
-            //if we're receiving an I2C reply, the first two bytes in our reply are the address and register
-            for( i = 2; i < len; ++i )
-            {
-                writer->WriteByte( argv_[i] );
-            }
-
-            caller_->I2cReplyReceived( caller_, ref new I2cCallbackEventArgs( argv_[0], argv_[1], writer->DetachBuffer() ) );
-        }
-        else
-        {
-            //if this isn't an I2C reply, all of the bytes received are relevant
-            for( i = 0; i < len; ++i )
-            {
-                writer->WriteByte( argv_[i] );
-            }
-
-            caller_->SysexMessageReceived( caller_, ref new SysexCallbackEventArgs( command_, writer->DetachBuffer() ) );
-        }
-    }
+    //version number and name array used with set/printFirmwareVersion
+    uint8_t firmwareVersionMajor;
+    uint8_t firmwareVersionMinor;
+    std::string *firmwareName;
 
-  private:
     //sysex-building
     const size_t MAX_SYSEX_LEN = 15;
     uint8_t _sys_command;
     uint8_t _sys_position;
 
-    //common buffer
+    //common buffer for outgoing messages, limited by the message size that can be read by Android's implementation of Firmata
     const size_t DATA_BUFFER_SIZE = 31;
-    std::unique_ptr<uint8_t> _data_buffer;
+    std::unique_ptr<uint16_t> _data_buffer;
```

Comment:
allows the user to send up to 14 bits of resolution, like firmata, instead of always being required to waste 8 bits (1 on first byte, 7 on second) for each byte of actual data sent. decoding would work the same on the other end, it would be up to the user to decode their bits properly (as it is now, if they wanted to send a 14 bit value they'd have to split it into two bytes on the UWP side, which would in turn be split into 4 actual bytes over the wire).